src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/BundleMapperTest.java

```diff
@@ -0,0 +1,57 @@
+package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Enumerations;
+import org.hl7.fhir.dstu3.model.HumanName;
+import org.hl7.fhir.dstu3.model.Practitioner;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.when;
+import static uk.nhs.digital.nhsconnect.lab.results.fixtures.FhirFixtures.generateRequester;
+import static uk.nhs.digital.nhsconnect.lab.results.fixtures.PathologyRecordFixtures.generatePathologyRecord;
+
+@ExtendWith(MockitoExtension.class)
+class BundleMapperTest {
+
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @InjectMocks
+    private BundleMapper bundleMapper;
+
+    @Test
+    void testMapPathologyRecordToBundle() {
+        when(uuidGenerator.generateUUID()).thenReturn("some-value-uuid").thenReturn("some-entry-uuid");
+
+        Practitioner generatedRequester = generateRequester("Dr Bob Hope", Enumerations.AdministrativeGender.MALE);
+        final Bundle bundle = bundleMapper.mapToBundle(generatePathologyRecord(generatedRequester));
+
+        assertNotNull(bundle.getMeta().getLastUpdated());
+        assertEquals("https://fhir.nhs.uk/STU3/StructureDefinition/ITK-Message-Bundle-1", bundle.getMeta().getProfile().get(0).asStringValue());
+        assertEquals("https://tools.ietf.org/html/rfc4122", bundle.getIdentifier().getSystem());
+        assertEquals("some-value-uuid", bundle.getIdentifier().getValue());
+        assertEquals(Bundle.BundleType.MESSAGE, bundle.getType());
+
+        assertEquals(1, bundle.getEntry().size());
+
+        Bundle.BundleEntryComponent bundleEntryComponentForRequesterResource = bundle.getEntry().get(0);
+        assertEquals("urn:uuid:some-entry-uuid", bundleEntryComponentForRequesterResource.getFullUrl());
+
+        Practitioner requester = (Practitioner) bundleEntryComponentForRequesterResource.getResource();
+        assertThat(requester.getName())
+                .hasSize(1)
+                .first()
+                .extracting(HumanName::getText)
+                .isEqualTo("Dr Bob Hope");
+        assertThat(requester.getGender().toCode())
+                .isEqualTo("male");
```

Comment:
To make this report as many failures as it could would be something like this:
```suggestion
        assertAll(
                () -> assertThat(bundle.getMeta().getLastUpdated()).isNotNull(),
                () -> assertThat(bundle.getMeta().getProfile().get(0).asStringValue())
                    .isEqualTo("https://fhir.nhs.uk/STU3/StructureDefinition/ITK-Message-Bundle-1"),
                
                () -> assertThat(bundle.getIdentifier().getSystem()).isEqualTo("https://tools.ietf.org/html/rfc4122"),
                () -> assertThat(bundle.getIdentifier().getValue()).isEqualTo("some-value-uuid"),
                
                () -> assertThat(bundle.getType()).isEqualTo(Bundle.BundleType.MESSAGE),
                
                () -> assertThat(bundle.getEntry())
                    .hasSize(1)
                    .first()
                    .extracting(Bundle.BundleEntryComponent::getFullUrl)
                    .isEqualTo("urn:uuid:some-entry-uuid"),
                () -> assertThat(bundle.getEntry())
                    .first()
                    .extracting(Bundle.BundleEntryComponent::getResource)
                    .extracting(Practitioner.class::cast)
                    .satisfies(requester -> assertAll(
                            () -> assertThat(requester.getName())
                                    .hasSize(1)
                                    .first()
                                    .extracting(HumanName::getText)
                                    .isEqualTo("Dr Bob Hope"),
                            () -> assertThat(requester.getGender().toCode())
                                    .isEqualTo("male")
                    ))
        );
```

which is... meh. Perhaps some half-way can be found which reports as many failures as possible, but retains better readability?